tests/unit_test/test_curve.py

```diff
@@ -0,0 +1,116 @@
+import numpy as np
+import pytest
+import unittest
+from beprof.curve import Curve
+
+
+class TestCurveInit(unittest.TestCase):
+    """
+    Testing Curve initialization and .x .y
+    """
+    def test_list_init(self):
+        simple_lists = [[-12, 1], [-1, 7], [0, 3], [3, 17]]
+        c = Curve(simple_lists)
+        assert list(c.x) == [e[0] for e in simple_lists]
+        assert list(c.y) == [e[1] for e in simple_lists]
+
+    def test_numpy_array_init(self):
+        numpy_array = np.array([[-12, 1], [-1, 7], [0, 3], [3, 17]])
+        c = Curve(numpy_array)
+        assert c.x.tolist() == numpy_array[:, 0].tolist()
```

Comment:
What about using numpy to compare numpy arrays, i.e. via http://docs.scipy.org/doc/numpy/reference/generated/numpy.equal.html ?
and then assertTrue